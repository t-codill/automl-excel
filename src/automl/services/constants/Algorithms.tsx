const algorithms = {
    classification: [
        "LogisticRegression",
        "SGD",
        "MultinomialNaiveBayes",
        "BernoulliNaiveBayes",
        "SVM",
        "LinearSVM",
        "KNN",
        "DecisionTree",
        "RandomForest",
        "ExtremeRandomTrees",
        "LightGBM",
        "GradientBoosting",
        "TensorFlowDNN",
        "TensorFlowLinearClassifier",
        "XGBoostClassifier"
    ],
    regression: [
        "ElasticNet",
        "GradientBoosting",
        "DecisionTree",
        "KNN",
        "LassoLars",
        "SGD",
        "RandomForest",
        "ExtremeRandomTrees",
        "LightGBM",
        "TensorFlowLinearRegressor",
        "TensorFlowDNN",
        "XGBoostRegressor"
    ]
};
export { algorithms as Algorithms };
