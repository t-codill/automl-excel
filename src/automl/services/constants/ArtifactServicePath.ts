export enum ArtifactServicePath {
    Model = "outputs/model.pkl",
    Scoring = "outputs/scoring_file_v_1_0_0.py",
    Conda = "outputs/conda_env_v_1_0_0.yml"
}
