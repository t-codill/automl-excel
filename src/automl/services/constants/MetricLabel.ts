import { AzureMachineLearningJasmineManagementModels } from "@vienna/jasmine";

const metricLabel: { [metric in AzureMachineLearningJasmineManagementModels.PrimaryMetric]: string } = {
    AUC_weighted: "Weighted Area Under The Curve",
    accuracy: "Accuracy",
    norm_macro_recall: "Normed Macro Recall",
    average_precision_score_weighted: "Weighted Average Precision",
    precision_score_weighted: "Weighted Precision",
    spearman_correlation: "Spearman Correlation",
    normalized_root_mean_squared_error: "Normalized Root Mean Squared Error",
    r2_score: "R2 Score",
    normalized_mean_absolute_error: "Normalized Median Absolute Error",
    normalized_root_mean_squared_log_error: "Normalized Root Mean Squared Log Error"
};

export { metricLabel as MetricLabel };
