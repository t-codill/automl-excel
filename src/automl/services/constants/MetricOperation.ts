import { AzureMachineLearningJasmineManagementModels } from "@vienna/jasmine";

type MetricOperation = "maximize" | "minimize";

const metricOperation: { [metric in AzureMachineLearningJasmineManagementModels.PrimaryMetric]: MetricOperation } = {
    AUC_weighted: "maximize",
    accuracy: "maximize",
    norm_macro_recall: "maximize",
    average_precision_score_weighted: "maximize",
    precision_score_weighted: "maximize",
    spearman_correlation: "maximize",
    normalized_root_mean_squared_error: "minimize",
    r2_score: "maximize",
    normalized_mean_absolute_error: "minimize",
    normalized_root_mean_squared_log_error: "minimize",
};

export { metricOperation as MetricOperation };
