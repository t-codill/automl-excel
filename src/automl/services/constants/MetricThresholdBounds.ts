import { AzureMachineLearningJasmineManagementModels } from "@vienna/jasmine";

const metricThresholdBounds: { [metric in AzureMachineLearningJasmineManagementModels.PrimaryMetric]: number[] } = {
    AUC_weighted: [0, 1],
    accuracy: [0, 1],
    norm_macro_recall: [0, 1],
    average_precision_score_weighted: [0, 1],
    precision_score_weighted: [0, 1],
    spearman_correlation: [-1, 1],
    normalized_root_mean_squared_error: [0, 10],
    r2_score: [-10, 1],
    normalized_mean_absolute_error: [0, 10],
    normalized_root_mean_squared_log_error: [0, 10],

};

export { metricThresholdBounds as MetricThresholdBounds };
