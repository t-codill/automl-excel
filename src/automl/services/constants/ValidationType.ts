
export enum ValidationType {
    crossValidation = 1,
    monteCarloValidation = 2
}
