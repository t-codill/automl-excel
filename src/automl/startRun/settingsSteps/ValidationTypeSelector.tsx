import { values } from "lodash";
import { Dropdown, IDropdownOption } from "office-ui-fabric-react";
import * as React from "react";
import { LabelWithCallout } from "../../components/Label/LabelWithCallout";
import { RunType } from "../../services/constants/RunType";
import { ValidationType } from "../../services/constants/ValidationType";

export interface IValidationTypeSelectorProps {
    validationType: ValidationType;
    jobType: RunType;
    onChange(type: ValidationType): void;
}

export interface IValidationTypeSelectorStates {
    options: IDropdownOption[];
}

export const validationTypeOptions = {
    CrossValidation: {
        key: ValidationType.crossValidation,
        text: "K-fold cross validation"
    },
    MonteCarlo: {
        key: ValidationType.monteCarloValidation,
        text: "Monte Carlo cross validation"
    }
};

export class ValidationTypeSelector extends React.Component<IValidationTypeSelectorProps, IValidationTypeSelectorStates> {
    constructor(props: IValidationTypeSelectorProps) {
        super(props);
        this.state = {
            options: this.getOptions()
        };
    }
    public componentDidUpdate(prevProp: IValidationTypeSelectorProps): void {
        if (this.props.jobType !== prevProp.jobType) {
            if (this.props.jobType === "forecasting") {
                this.props.onChange(ValidationType.crossValidation);
            }
            this.setState({ options: this.getOptions() });
        }
    }
    public render(): React.ReactNode {
        const dropdownId = "validationType";

        return <div className="ms-Grid" dir="ltr">
            <div className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6">
                <LabelWithCallout required={false} htmlFor={dropdownId} labelText="Validation type">
                    The type of validation, forecasting only supports K-fold cross validation
                </LabelWithCallout>
            </div>
            <div className="ms-Grid-col ms-sm12 ms-md6 ms-lg6 ms-xl6">
                <Dropdown
                    placeholder="Select an validation type"
                    ariaLabel="Validation type"
                    options={this.state.options}
                    onChange={this.handleOptionChange}
                    defaultSelectedKey={this.props.validationType}
                    disabled={this.props.jobType === "forecasting"}
                />
            </div>
        </div>;
    }
    public handleOptionChange = (_event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption): void => {
        if (option) {
            this.props.onChange(option.key as ValidationType);
        }
    }

    private getOptions(): IDropdownOption[] {
        if (this.props.jobType === "forecasting") {
            return [validationTypeOptions.CrossValidation];
        }
        return values(validationTypeOptions);
    }
}
