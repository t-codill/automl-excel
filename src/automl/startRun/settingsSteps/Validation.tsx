import * as React from "react";
import { LabelWithCallout } from "../../components/Label/LabelWithCallout";
import { RunType } from "../../services/constants/RunType";
import { ValidationType } from "../../services/constants/ValidationType";
import { CrossValidationNumberInput } from "./CrossValidationNumberInput";
import { MonteCarloInput } from "./MonteCarloInput";
import { ValidationTypeSelector } from "./ValidationTypeSelector";

export interface IValidationProps {
    jobType: RunType;
}
export interface IValidationState {
    validationType: ValidationType;
}

export class Validation extends React.Component<IValidationProps, IValidationState> {
    constructor(props: IValidationProps) {
        super(props);
        this.state = {
            validationType: ValidationType.crossValidation
        };
    }
    public render(): React.ReactNode {
        return <>
            <LabelWithCallout required={false} htmlFor={"validation"} labelText="Validation">
                Choose validation type
            </LabelWithCallout>
            <div>
                <ValidationTypeSelector
                    jobType={this.props.jobType}
                    validationType={this.state.validationType} onChange={this.handleTypeChange} />
                {
                    this.state.validationType === ValidationType.crossValidation &&
                    <CrossValidationNumberInput />
                }
                {
                    this.state.validationType === ValidationType.monteCarloValidation &&
                    <MonteCarloInput />
                }
            </div>
        </>;
    }
    public handleTypeChange = (validationType: ValidationType): void => {
        this.setState({ validationType });
    }
}
