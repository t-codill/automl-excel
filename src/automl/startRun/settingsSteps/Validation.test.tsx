import { shallow, ShallowWrapper } from "enzyme";
import * as React from "react";
import { ValidationType } from "../../services/constants/ValidationType";
import { IValidationProps, IValidationState, Validation } from "./Validation";
import { ValidationTypeSelector } from "./ValidationTypeSelector";

describe("Cross Validation Input", () => {
    let tree: ShallowWrapper<IValidationProps, IValidationState>;
    beforeAll(() => {
        tree = shallow(
            <Validation
                jobType="classification"
            />
        );
    });
    it("should render with valid props", () => {
        expect(tree)
            .toMatchSnapshot();
    });
    it("should render with valid props", () => {
        tree.find(ValidationTypeSelector)
            .props()
            .onChange(ValidationType.monteCarloValidation);
        expect(tree.state().validationType)
            .toBe(ValidationType.monteCarloValidation);
    });
});
