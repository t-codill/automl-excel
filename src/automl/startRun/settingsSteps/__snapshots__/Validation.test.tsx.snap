// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cross Validation Input should render with valid props 1`] = `
<Fragment>
  <LabelWithCallout
    htmlFor="validation"
    labelText="Validation"
    required={false}
  >
    Choose validation type
  </LabelWithCallout>
  <div>
    <ValidationTypeSelector
      jobType="classification"
      onChange={[Function]}
      validationType={1}
    />
    <crossValidationNumberInput />
  </div>
</Fragment>
`;
